propkatraj/propkatraj.py

```diff
@@ -2,22 +2,239 @@
+    input residues to residues numbered 1 through to 10::
+
+      pkatraj = PropkaTraj(u, select="resnum 1-10")
+
+    Alternatively, you can create an atomgroup beforehand and pass `None` to
+    ``select``::
+
+      ag = u.select_atoms('resnum 1-10")
+      pkatraj = PropkaTraj(ag, select=None)
+
+    Once the :class:`PropkaTraj` object has been created, we then use the
+    :meth:`run` to analyse the trajectory::
+
+      pkatraj.run()
+
+    In some cases, especially considering that calling propka can be time
+    consuming for large systems, you may not want to analyse every frame.
+    You can pass the ``start``, ``stop``, and ``step`` arguments to :meth:`run`
+    in order to alter the starting frame, final frame and frame sampling
+    frequency. Please note that frame numbers are zero formatted, and the
+    ``stop`` keyword is exclusive, as is the default in
+    :class:`MDAnalysis.AnalysisBase` objects. For example, in order to sample
+    every two frames from the first 10 frames::
+
+      pkatraj.run(start=0, stop=10, step=2)
+
+    You can also set the ``verbose`` keyword in order to have a nice progress
+    bar of the analysis progress::
+
+      pkatraj.run(verbose=True)
+
+    Once completed, the resulting timeseries of predicted pKas per residue is
+    made available as a :class:`pandas.DataFrame` under :attr`PropkaTraj`.pkas.
+    For example if one wanted to get a summary of the statistics of the
+    timeseries::
+
+      pkatraj.pkas.describe()
+
+    If one wanted to plot per residue boxplots of the timeseries data and save
+    it as a file `pKa-plot.png` (note: this requires the ``matplotlib`` and
+    ``seaborn`` packages which are not installed by default with
+    ``propkatraj``)::
+
+      import matplotlib.pyplot as plt
+      import seaborn as sns
+
+      fig = plt.figure(figsize=(28, 8))
+      ax = fig.add_subplot(1, 1, 1)
+      sns.boxplot(data=pkatraj.pkas, ax=ax)
+      ax.set_xlabel('residue number')
+      ax.set_ylabel(r'p$K_a$')
+      fig.savefig('pKa-plot.png')
+
+
+    .. versionadded:: 1.0.3
+    """
+    def __init__(self, atomgroup, select='protein', skip_failure=False,
+                 **kwargs):
+        if select is not None:
+            self.ag = atomgroup.select_atoms(select).atoms
+        else:
+            self.ag = atomgroup.atoms
+
+        # Issue #23 (keep until the PDBWriter is fixed)
+        if len(self.ag.select_atoms('not protein')) > 0:
+            wmsg = ("Non protein atoms passed to propka 3.1.\n MDAnalysis' "
+                    "PDBWriter does not currently write non-standard residues "
+                    "correctly as HETATM records and this may lead to "
+                    "incorrect pKa predictions.\n"
+                    "See https://github.com/Becksteinlab/propkatraj/issues/24 "
+                    " for more details")
+            warnings.warn(wmsg)
+
+        # TODO: probably needs to be a temporary directory instead
```

Comment:
I agree but I don't remember if propka can be easily convinced to write to arbitrary paths. It might require the equivalent of a `cd`.

Reply:
So looking into this more, it seems that propka3.1 will always write out the input file and `.pka` file to the current working directory, so having a temporary directory doesn't fix things. It seems like things might be a bit different in 3.2 (if anything we can avoid writing the input file). I've removed the comment for now.